### PR TITLE
Reminder notifications

### PR DESCRIPTION
**Pull request by nzeager**
### **1. Targeted Issue** ###

[Ticket #93 15/60 min notifications BE](https://trello.com/c/Mfoy9Axe): Adding backend functionality to send users 15 and 60 minute notifications before a session begins.

### **2. Overview of Solution** ###

We used Redis, Celery, and Celery Beat to check every 5 minutes if we are within 10-15 or 55-60 minutes of a session. If so we call the appropriate function we set up in models.py to email mentors and mentees to remind them how soon their session is starting. celery.py uses Celery Beat to say we want to run this check every 5 minutes. tasks.py uses Celery and checks if the session is in the appropriate time range.

### **3. Tools Used** ###

We used Redis, Celery, and Celery Beat.

### **4. Testing Strategy** ###

Run the local server, Redis, Celery, and Celery Beat (see README for instructions on how to do that) and set up sessions on my local server. Then check to see if the emails are sent at the appropriate times.

### **5. Future Implications** ###

We'll want to add testing to the code we added in model.py down the line (added TODO comments).

### **6. Screenshots** ###

n/a

### **7. Code Reviewers** ###

@GitLukeW @esparr @jeromecox This was a team effort, thanks for everyone's contributions!

### CODE REVIEW

**Review by GitLukeW — CHANGES_REQUESTED** [review]
Looks like it has a conflict in the pipfile. We will need to get that fixed before merging.

**Comment by nzeager** [comment]
Need to merge some updates from the main branch. Please hold off on reviewing until noted.

File: config/celery.py
Comment (esparr) [inline config/celery.py:8]: Let's look into how to automate the production versus development environments for celery. Here is the Render documentation for Celery: https://render.com/docs/deploy-celery

File: team_production_system/models.py
Comment (esparr) [inline team_production_system/models.py:262]: Let's add a comment here for unit testing. Ex:` #TODO: Needs unit testing`

File: team_production_system/models.py
Comment (esparr) [inline team_production_system/models.py:279]: Same to do comment here as well for testing.

**Review by esparr — APPROVED** [review]
Looks great! Excited about getting this implemented. I added a few comments and approved.

File: team_production_system/models.py
Comment (nzeager) [reply team_production_system/models.py:280]: Added comment

File: team_production_system/models.py
Comment (nzeager) [reply team_production_system/models.py:263]: Added comment

**Review by GitLukeW — APPROVED** [review]
Thank you!

**Pull request merged**
